functions/src/api/auth/assertValidToken.spec.ts

```diff
@@ -0,0 +1,46 @@
+// tslint:disable-next-line
+import '../../testUtils/mockFirebaseAdminAndFunctions'
+import { firestoreGet } from '../../testUtils/firestoreStub'
+import { JWTData } from '../../types/JWT'
+import { assertValidToken } from './assertValidToken'
+import { ForbiddenError } from '../errors/HttpError'
+
+describe('assertValidToken', () => {
+    const token = 'tokenValue'
+    const jwtData: JWTData = {
+        uid: '69',
+        iat: 1544455,
+    }
+
+    it('should not throw exception because token is valid in DB', async () => {
+        const snapshot = {
+            data: () => ({
+                tokens: {
+                    [token]: {
+                        valid: true,
+                    },
+                },
+            }),
+            exists: true,
+        }
+        firestoreGet.mockImplementation(() => Promise.resolve(snapshot))
+
+        const result = await assertValidToken(jwtData, token)
+
+        expect(result).toBeUndefined()
+    })
+
+    it('should throw exception because token is NOT valid in DB', async () => {
+        const snapshot = {
+            data: () => ({}),
+            exists: true,
+        }
+        firestoreGet.mockImplementation(() => Promise.resolve(snapshot))
+
+        try {
+            await assertValidToken(jwtData, token)
+        } catch (error) {
```

Comment:
if the code presents an error, and the `assertValidToken` function does not correctly raise an exception, wouldn't the test pass with a false positive ? (I think we miss a failing `expect` after the `await assert...`)

Reply:
if it's the case, the error will not be an instance of ForbiddenError as it's specified manually from the code itself.

Reply:
What I meant is: we are testing that the function raises an exception. What if the function does not instead ?

```typescript
try {
  await ...
  // what if the code execution goes here, since the function did not raise an exception as expected ? Will the test fail or pass ?
} catch(e) {
   expect(...) // We test the function raises an exception here, as expected
}
```

Reply:
the test will fail because the expect from the catch was not called.